app/src/lib/git/squash.ts

```diff
@@ -29,6 +29,12 @@ export async function squash(
       throw new Error('[squash] No commits provided to squash.')
     }
 
+    if (toSquash.find(c => c.sha === squashOnto.sha) !== undefined) {
```

Comment:
Any reason for not reusing the new `toSquashShas` here? 🤔 
```suggestion
    if (toSquashShas.has(squashOnto.sha)) {
```

Reply:
Hmm.. logically I am in favor of cutting out before doing the logic to map through them... but in the ideal scenario (which should be all of them since we are controlling it :D) I can see this would be less performant.